### PR TITLE
MM-23707 Add support for new helm repos

### PR DESCRIPTION
**Pull request by stylianosrigas**
This PR:

- Adds support to add new helm repos
- Adds flags for cert manager and public nginx helm charts
- Fixes utility group test
- Increases the number of attempts to deploy clusterissuer to cover edge cases

### CODE REVIEW

File: internal/provisioner/helm_utils.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ type helmDeployment struct {
 	logger          log.FieldLogger
 }
 
-func installHelm(kops *kops.Cmd, logger log.FieldLogger) error {
+func installHelm(kops *kops.Cmd, repos map[string]string, logger log.FieldLogger) error {
```
Comment (gigawhitlocks) [inline internal/provisioner/helm_utils.go:38]: :+1:

File: internal/provisioner/helm_utils.go [hunk 2]
```diff
@@ -53,6 +53,13 @@ func installHelm(kops *kops.Cmd, logger log.FieldLogger) error {
 	}
 
 	logger.Info("Updating all Helm repos.")
+	for repoName, repoURL := range repos {
+		err = helmRepoAdd(repoName, repoURL, logger)
+
```
Comment (gigawhitlocks) [inline internal/provisioner/helm_utils.go:58]: nit: extra newline here

Comment (stylianosrigas) [reply internal/provisioner/helm_utils.go]: thanks

File: cmd/cloud/cluster.go [hunk 3]
```diff
@@ -23,11 +23,15 @@ func init() {
 	clusterCreateCmd.Flags().String("prometheus-version", model.PrometheusDefaultVersion, "The version of Prometheus to provision. Use 'stable' to provision the latest stable version published upstream.")
 	clusterCreateCmd.Flags().String("fluentbit-version", model.FluentbitDefaultVersion, "The version of Fluentbit to provision. Use 'stable' to provision the latest stable version published upstream.")
 	clusterCreateCmd.Flags().String("nginx-version", model.NginxDefaultVersion, "The version of Nginx to provision. Use 'stable' to provision the latest stable version published upstream.")
+	clusterCreateCmd.Flags().String("publicnginx-version", model.PublicNginxDefaultVersion, "The version of Public Nginx to provision. Use 'stable' to provision the latest stable version published upstream.")
```
Comment (cpanato) [inline cmd/cloud/cluster.go:26]: maybe `public-nginx-version`

File: cmd/cloud/cluster.go [hunk 4]
```diff
@@ -23,11 +23,15 @@ func init() {
...
+	clusterCreateCmd.Flags().String("certmanager-version", model.CertManagerDefaultVersion, "The version of Cert Manager to provision. Use 'stable' to provision the latest stable version published upstream.")
```
Comment (cpanato) [inline cmd/cloud/cluster.go:27]: `cert-manager-version` maybe as well?

File: internal/provisioner/utility_group.go [hunk 5]
```diff
@@ -118,8 +118,13 @@ func (group utilityGroup) CreateUtilityGroup() error {
 		return errors.Wrap(err, "failed to deploy Cert Manager manifests")
 	}
 
+	// List of repos to add during helm setup
+	var helmRepos = map[string]string{
+		"jetstack": "https://charts.jetstack.io",
```
Comment (cpanato) [inline internal/provisioner/utility_group.go:123]: I will add the stable as well just to be explicitly

File: cmd/cloud/cluster.go (hunk 3 above)
Comment (stylianosrigas) [reply cmd/cloud/cluster.go]: fixed thanks

File: cmd/cloud/cluster.go (hunk 4 above)
Comment (stylianosrigas) [reply cmd/cloud/cluster.go]: fixed thanks

File: internal/provisioner/utility_group.go (hunk 5 above)
Comment (stylianosrigas) [reply internal/provisioner/utility_group.go:123]: added stable :)

File: cmd/cloud/cluster.go [hunk 6]
```diff
@@ -314,6 +318,8 @@ func processUtilityFlags(command *cobra.Command) map[string]string {
 	prometheusVersion, _ := command.Flags().GetString("prometheus-version")
 	fluentbitVersion, _ := command.Flags().GetString("fluentbit-version")
 	nginxVersion, _ := command.Flags().GetString("nginx-version")
+	publicnginxVersion, _ := command.Flags().GetString("public-nginx-version")
```
Comment (cpanato) [inline cmd/cloud/cluster.go:321]:
```suggestion
	publicNginxVersion, _ := command.Flags().GetString("public-nginx-version")
```

File: cmd/cloud/cluster.go [hunk 7]
```diff
@@ -314,6 +318,8 @@ func processUtilityFlags(command *cobra.Command) map[string]string {
...
+	certmanagerVersion, _ := command.Flags().GetString("cert-manager-version")
```
Comment (cpanato) [inline cmd/cloud/cluster.go:322]:
```suggestion
	certManagerVersion, _ := command.Flags().GetString("cert-manager-version")
```
